EightfoldAI/octuple
# fix: pagination: Issues with visiblePagerCountSize

## SUMMARY:

- Currently, we show at least 10 pages before we show the quick previous and quick next arrows. This is not suitable for smaller UIs with limited width like a Chrome Extension. 

![image](https://user-images.githubusercontent.com/106535945/235666033-2b4c9383-2c26-4584-8732-ac25d2e990f1.png)

**Solution:** SHORT_LIST_THRESHOLD causes this. To address this we allow to change this limit to 8 pages in case `visiblePagerCountSize` is set to 5

- When `visiblePagerCountSize` is 5, there is also another bug where some of the page numbers are not visible even though they are active. See the 5th page number in the below video

https://user-images.githubusercontent.com/106535945/235666857-58341b86-f7ec-44ff-8a14-6b2cb7c8709b.mov

Solution: EDGE_LIST_THRESHOLD causes this. To address this we allow to change this limit to display 3 pages between Quick Previous and Quick Next buttons in case `visiblePagerCountSize` is set to 5

- Clean up

Removed an option which allowed  `visiblePagerCountSize` to be 3 as it doesn't make any visual sense in terms of UI/UX to the user, nor do we use it in any of our products

## JIRA TASK (Eightfold Employees Only):

https://eightfoldai.atlassian.net/browse/ENG-51566

## CHANGE TYPE:


## TEST COVERAGE:


## TEST PLAN:

Test case 1:
- Switch to the branch of PR
- Run storybook
- Go to pagination
- Set visiblePagerCountSize to small, pageSize to 10, total to 80
- 8 pages should be displayed

<img width="430" alt="Screenshot 2023-05-02 at 6 14 55 PM" src="https://user-images.githubusercontent.com/106535945/235669764-63d7bec6-cda2-44e5-998a-8359a51a1750.png">

Test case 2:

- Switch to the branch of PR
- Run storybook
- Go to pagination
- Set visiblePagerCountSize to small, pageSize to 10, total to 81
- 9 pages should be displayed with Quick Previous and Quick Next

<img width="430" alt="Screenshot 2023-05-02 at 6 15 18 PM" src="https://user-images.githubusercontent.com/106535945/235669845-1e31da0d-81fd-4a80-854b-e5cb5e455715.png">

Test case 3:

- Switch to the branch of PR
- Run storybook
- Go to pagination
- Set visiblePagerCountSize to small, pageSize to 10, total to 81
- Switch to 5th page by clicking on next button only
- You should see the 5th page active.
